RFC-0003-arrow-flight-connector.md

```diff
@@ -71,6 +72,36 @@ Flight descriptor will be created from a command of byte array. The concrete imp
 
 ![Class diagram](arrow-flight-connector/Class-diagram.png)
 
+
+## Prestissimo Implementation
+A single Prestissimo connector will handle all concrete instances of the Java Arrow Flight connector. This is possible because the prestissimo connector does not have to deal with metadata fetch and discovery. Each split wraps a ticket and a list of locations, which is all[^1] that is required to fetch data from the Flight Service in an implementation-agnostic way.
+
+[^1]: Authentication is one exception which cannot be handled in a truly generic manner. A solution is proposed below.
+
+### Arrow Record Batch to Velox vectors
+Arrow arrays and Velox vectors binary compatible with the Arrow C Data Interface.
+- Velox provides bridge methods to convert Velox vectors to-and-from from the C Interface [(Link)](https://github.com/facebookincubator/velox/blob/main/velox/vector/arrow/Bridge.h)
+- The Arrow SDK provides bridge method to convert Arrow arrays to-and-from the C Interface [(Link)](https://arrow.apache.org/docs/cpp/api/c_abi.html)
+
+The Arrow Flight SDK returns data in the form of Arrow Arrays, which can be converted into Velox vectors by using the provided bridge methods, using the Arrow C Interface as an intermediate representation. In most cases, these are cheap operations that don't involve copying or translating the actual data buffers.
+
+### Plugin-style authentication
+The worker needs to authenticate with the Flight server to redeem the Flight Ticket. This cannot be handled in a truly generic manner since the Arrow Flight protocol allows a lot of freedom to implement custom authentication strategies. Each implementaion of a flight service may require custom code to handle authentication.
+
+We propose to solve this by adding plugin-style authenticators to the prestissimo connectors. A Flight catalog can optionally specify an authenticator to use:
+```properties
+connector.name=arrow-flight
+flight.auth.name=simple-user-pass
```

Comment:
Are these configs only for the Prestissimo connector?

Reply:
The `flight.auth.name` property is technically only for Prestissimo, since Presto can already determine the authenticator type using `connector.name`. The other authenticator specific properties like `flight.auth.username` will be the same as the ones used for Presto (they need not start with `flight.auth`, but I think that makes things clearer)

Reply:
But we will only have the base flight module in the OSS PR. So how does implementation specific properties come into picture here?

Reply:
Ideally, we would send a concrete implementation of the base flight module as well (like Flight SQL). This would be better than just sending the base module because the base module by itself can't really be used unless a sub-module is created.

To be blunt, if we just send the base module, there's no proof that it works.

Even if we don't send a concrete module, we still need to worry about how we would allow others to implement their own authentication strategies

Reply:
As requested, updated the comment below to make it clear that these are examples.